.github/workflows/rebase_latest_version.yml

```diff
@@ -0,0 +1,38 @@
+name: Auto rebase the latest version branch with main
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  synchronise_version_branch:
+    name: Synchronize the latest branch
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
+        with:
+          fetch-depth: 0
+
+      - name: Get branch for the latest version
+        run: |
+          echo "DATA_LATEST_VERSION=\
+          $(git for-each-ref --format='%(refname:short)' refs/remotes/origin \
+          | grep -Po '(?<=origin/)v\d.\d(.\d)?$' \
+          | sort -V | tail -n 1\
+          )" >> $GITHUB_ENV
+
+      - name: Rebase and push
+        run: |
+          git switch $DATA_LATEST_VERSION
+          git rebase main
+          git push --force-with-lease
+
+      - name: Slack Notification
+        if: ${{ always() && job.status == 'failure' }}
+        continue-on-error: true
+        uses: rtCamp/action-slack-notify@4e5fb42d249be6a45a298f3c9543b111b02f7907
+        env:
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_MESSAGE: "backward-compat-data version auto-rebase failed: (${{ env.ACTION_RUN_URL }})"
```

Comment:
could we have the PR number or even a link embedded in the notification such that we can take action more quickly in case of failure ?

Reply:
I think it's what `${{ env.ACTION_RUN_URL }}` is for ?

I copied the slack part from tfhe-rs, the same pattern is used in most of our workflows

Reply:
ah wait, no I'm mistaken, we would need the branch that failed to rebase, not a PR